src/en/users-auth.md

```diff
@@ -24,22 +25,90 @@ juju bootstrap aws mycontroller
 juju change-user-password
 ```
 
-See [Creating a controller](./controllers-creating.html) for details on
-controller creation.
+See [Creating a controller][controllers-creating] for details on controller
+creation.
 
+## Credentials
 
-## Credentials and SSH keys
-
-*Credentials* managed in Juju are related to the accessing of the chosen
+Credentials managed in Juju are related to the accessing of the chosen
 cloud-backed resource. They are not related to Juju users themselves in any
 way. Credentials added to Juju remain local to the system user (on Ubuntu:
 `~/.local/share/juju/credentials.yaml`).
 
-See [Cloud credentials](./credentials.html) for more on this topic.
+See [Cloud credentials][credentials] for more on this topic.
+
+## SSH access
+
+SSH access is managed on a per-model basis. That is, if a public key is added
+to a model then that key is placed on all machines (present and future) in that
+model.
+
+Each Juju machine provides a user account named 'ubuntu' and it is to this
+account that public keys are added when using the Juju SSH commands (
+`juju add-ssh-key` and `juju import-ssh-key`). Because this user is effectively
+the 'root' user (passwordless sudo privileges), the granting of SSH access must
+be done with due consideration.
+
+### Native ssh
+
+When using the native (OpenSSH) `ssh` client, if one's public key has been
+installed into a model, then, as expected, a connection to the 'ubuntu' user
+account can be made. All that is needed is the corresponding keypair and
+adequate network connectivity. 
+
+For example, to connect to a machine with an IP address of 10.149.29.143:
+
+```bash
+ssh ubuntu@10.149.29.143
+```
+
+### Juju ssh
+ 
```

Comment:
So, the problem I have here isn't with the words but rather with the whole concept here. Juju provides additional 'security', except it doesn't, because if you don't have the correct access rights, you can just use a standard ssh client and get access anyhow

Reply:
Perhaps this needs to be highlighted  in the first place - if you want Juju to have better security, you shouldn't add any keys at all, and perhaps this should actually be the main thrust of this section - start with Juju first, then deal with adding keys and direct access later, with a warning suggesting this leaves the machines open to unfettered access.

Reply:
Understood, except for this:

> if you want Juju to have better security, you shouldn't add any keys at all

A key *always* needs to be added. The only thing that is optional, in terms of allowing an ultimate connection, is 'admin' access at the model level. If it is granted, the user can use either `juju ssh` or a standard client. If it isn't, the user can use a standard client.

Given the extra steps a user needs to go through in order to use `juju ssh` (one time stuff like install Juju and register the user, but also needing to always log in to the controller, as well as reminding the Juju admin to grant 'admin' access; which probably confers other rights that the user should not have like destroying the model) I would think that the majority of users will opt for direct access. I am going to open a Juju bug on the matter (to somehow enforce `juju ssh`).

Reply:
https://bugs.launchpad.net/juju/+bug/1719682

Reply:
To clarify, you don't need to add a key - juju automatically adds it's own keys. that seems to be the only way of ensuring that people don't just use direct access all the time. I.e., if the admin user disables adding keys.

Reply:
addendum - i'm not sure if that holds true when you are using a juju client from more than one machine (e.g. multi user on multiple machines)

Reply:
So I confirmed that my understanding, as described in this PR, is correct.

Re your suggestion to "list keys post controller-creation", the admin's keys cannot be listed via `juju ssh-keys`. They are filtered out. :boar: